app/controllers/api_controller/parser.rb

```diff
@@ -2,6 +2,11 @@ class ApiController
   module Parser
     def parse_api_request
       @req = RequestAdapter.new(request, params)
+
+      if !collection_option?(:arbitrary_resource_path) && @req.collection_path_elements.length > 4
+        raise BadRequestError,
```

Comment:
I think the router would previously have returned a 404 for this. Was it intentional to change it to a 400?